presto-main/src/main/java/com/facebook/presto/operator/aggregation/builder/InMemoryHashAggregationBuilder.java

```diff
@@ -326,9 +327,9 @@ public int getCapacity()
     private boolean updateMemoryWithYieldInfo()
     {
         long memorySize = getSizeInMemory();
-        if (partial) {
+        if (partial && maxPartialMemory.isPresent()) {
```

Comment:
Can you add a comment stating that when there is not partial memory limit set, any memory usage is considered user memory?